### PR TITLE
Be resilient to not having a containing type for a backing field.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
fixes https://devdiv.visualstudio.com/DevDiv/Managed%20Languages%20IDE/_workitems?id=362923&_a=edit

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide This would be for RTW.  Please review asap.

**Comment by CyrusNajmabadi** [comment]
Note: i'm not sure what would cause this.  As far as I knew, if you had a field symbol, then it had to have a containing type.  But we're seeing crashes here in the wild, so this just makes us resilient to that situation.

**Comment by Pilchie** [comment]
Tagging @MattGertz for approval.

File: src/Features/Core/Portable/ReplacePropertyWithMethods/AbstractReplacePropertyWithMethodsService.cs [hunk 1]
```diff
@@ -24,11 +24,17 @@ internal abstract class AbstractReplacePropertyWithMethodsService<TIdentifierNam
 
         protected static SyntaxNode GetFieldReference(SyntaxGenerator generator, IFieldSymbol propertyBackingField)
         {
-            var through = propertyBackingField.IsStatic
-                ? generator.TypeExpression(propertyBackingField.ContainingType)
-                : generator.ThisExpression();
+            var memberName = generator.IdentifierName(propertyBackingField.Name);
+            if (propertyBackingField.IsStatic)
+            {
+                return propertyBackingField.ContainingType == null
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/ReplacePropertyWithMethods/AbstractReplacePropertyWithMethodsService.cs]: Can we at least add a comment that we don't understand why this is possible? Has the compiler team looked to make sure there's no obvious races here?

**Comment by MattGertz** [comment]
(Will need template.)

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @gafter We're seeing a crash where an IFieldSymbol has a null 'ContainingType.  Are there any scenarios where you know that this might happen?  If not, do we have some sort of worrying race-condition that would make this happen in the wild?

**Comment by CyrusNajmabadi** [comment]
> Customer scenario

Customer runs the "ReplacePropertyWithMethods" refactoring.  They encounter a crash.

> Bugs this fixes:

https://devdiv.visualstudio.com/DevDiv/Managed%20Languages%20IDE/_workitems?id=362923&_a=edit

> Workarounds, if any

 Customer doesn't invoke the refactoring.

> Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

This shuts off an entire feature for customers.

> Risk

none.

> Performance impact

none.

> Is this a regression from a previous update?

No.  This is a new feature.

> Root cause analysis:

This actually seems like this might be a bug in the compiler.  It's the understanding of the IDE team that the compiler should not produce 'Field Symbols' that don't have a 'Containing Type'.  However, we've run into a real world crash where that's not the case.  Regardless, we'd like to not make our feature crash when it encounters this sort of strange circumstance.  We've also roped in the compiler team to try to shed some light on the underlying issue and if it means there is a bug in their side of things.

> How did we miss it? What tests are we adding to guard against it in the future?

This is possibly a race condition deep in teh compiler code.  If we discover that it is, we'll have to do a proper audit to make sure the same issue doesn't exist elsewhere.

> How was the bug found?

Customer reported.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz @natidea

**Pull request merged**